### ISSUE TITLE
Removal of defined registers from microcode

### ISSUE THREAD

**Issue by pseudophpt** [pseudophpt/pseultra#11]
Currently, the scalar unit instructions in microcode use register mnemonics defined in `regdef.h`. However, these are not applicable to microcode, so they should be `#undef`'d and numbered registers (`r1`, `r2`, etc. etc.) should be used instead.

**Comment by pseudophpt** [comment]
The following register map can be used for scalar *and* vector registers:
```
r0: zero register
r1-r15: Preserved values, including constants, and values that are constantly written to or read from
r16-r23: Command level values, which can be safely modified in any command handler
r24-r30: Routine level values, which are used to pass arguments to routines, to perform operations within routines, and to return a value from a routine
r31: Return address for routines
```

**Comment by pseudophpt** [comment]
Scalar register definitions implemented as of commit dbefb9fb70e652ce79ffaaa820a8efbba28babdc.

**Comment by pseudophpt** [comment]
Registers have been been organized in the following fashion by commits 8016007a186d234fe1373f8ea9a9e2b4b5e71e07 and 87f126f389d95e9cab8b99d42393040e564adfdc:
```
s0-s8: Preserved values
t0-t9: Command level values
a0-a3, v0-v1: Routine level values
```
All that is left to do for this issue is to change the registers to their appropriate mapped locations as described in the above posts.

**Comment by pseudophpt** [comment]
Fixed in commit d6f0fae3e329d6ce9681229dc0381f3c62f7f6be.
